apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationFooter.tsx

```diff
@@ -138,145 +159,320 @@ export const ApplicationDetailsFooter = ({
     [dispatch, entity],
   );
 
+  const handleOpenSharing = useCallback(() => {
+    dispatch(
+      ShareActions.share({
+        featureType: FeatureType.Application,
+        resourceId: entity.id,
+      }),
+    );
+  }, [dispatch, entity.id]);
+
   const isApplicationsSharingEnabled = useAppSelector((state) =>
     SettingsSelectors.isFeatureEnabled(state, Feature.ApplicationsSharing),
   );
 
+  const link = useMemo(
+    () =>
+      `${window.location.origin}/${PageType.Marketplace}?${MarketplaceQueryParams.model}=${entity.reference}`,
+    [entity.reference],
+  );
+
+  const handleCopy = useCallback(() => {
+    if (!navigator.clipboard) return;
+    navigator.clipboard.writeText(link);
+    dispatch(UIActions.showSuccessToast(t('Link copied!')));
+  }, [dispatch, link, t]);
+
+  const menuItems: DisplayMenuItemProps[] = useMemo(
+    () => [
+      {
+        name: t('Copy link'),
+        dataQa: 'application-copy-link',
+        display: isPublicApp && isSmallScreen,
+        Icon: IconLink,
+        onClick: (e: React.MouseEvent) => {
+          handleCopy();
+          e.preventDefault();
+          e.stopPropagation();
+        },
+      },
+      {
+        name: t(getFunctionTooltip(entity)),
+        dataQa: 'status-change',
+        display: isExecutable && isCodeAppsEnabled,
+        disabled: playerStatus === SimpleApplicationStatus.UPDATING,
+        Icon: PlayerContextIcon,
+        iconClassName: PlayerContextIconClasses[playerStatus],
+        onClick: (e: React.MouseEvent) => {
+          e.stopPropagation();
+          handleUpdateFunctionStatus();
+        },
+      },
+      {
+        name: t('Share'),
+        dataQa: 'share',
+        display: isMyApp && isApplicationsSharingEnabled && isSmallScreen,
+        Icon: IconUserShare,
+        onClick: (e: React.MouseEvent) => {
+          handleOpenSharing();
+          e.stopPropagation();
+        },
+      },
+      {
+        name: t('Unshare'),
+        dataQa: 'unshare',
+        display: !!entity.sharedWithMe && isApplicationsSharingEnabled,
+        Icon: IconUserUnshare,
+        onClick: (e: React.MouseEvent) => {
+          handleOpenUnshare();
+          e.stopPropagation();
+        },
+      },
+      {
+        name: t('Delete'),
+        dataQa: 'delete',
+        display: isMyApp,
+        disabled: isModifyDisabled,
+        Icon: IconTrashX,
+        onClick: (e: React.MouseEvent) => {
+          e.stopPropagation();
+          onDelete?.(entity);
+        },
+      },
+      {
+        name: t('Publish'),
+        dataQa: 'publish',
+        display: isApplicationId(entity.id) && isMyApp,
+        Icon: IconWorldShare,
+        onClick: (e: React.MouseEvent) => {
+          e.stopPropagation();
+          onPublish?.(entity, PublishActions.ADD);
+        },
+      },
+      {
+        name: t('Unpublish'),
+        dataQa: 'unpublish',
+        display: isApplicationId(entity.id) && isPublicEntity,
```

Comment:
`isPublicEntity` already checks it
```suggestion
        display: isPublicEntity,
```